subxt/src/events/events_type.rs

```diff
@@ -480,222 +388,15 @@ fn decode_raw_event_details<T: Config>(
-                            Err(EventsDecodingError::InvalidCompactType(
-                                "Composite type must have a single field".into(),
-                            )
-                            .into())
-                        }
-                    }
-                }
-                _ => {
-                    Err(EventsDecodingError::InvalidCompactType(
-                        "Compact type must be a primitive or a composite type".into(),
-                    )
-                    .into())
-                }
-            }
-        }
-        TypeDef::BitSequence(bitseq) => {
-            let bit_store_def = types
-                .resolve(bitseq.bit_store_type().id())
-                .ok_or(MetadataError::TypeNotFound(type_id))?
-                .type_def();
-
-            // We just need to consume the correct number of bytes. Roughly, we encode this
-            // as a Compact<u32> length, and then a slice of T of that length, where T is the
-            // bit store type. So, we ignore the bit order and only care that the bit store type
-            // used lines up in terms of the number of bytes it will take to encode/decode it.
-            match bit_store_def {
-                TypeDef::Primitive(TypeDefPrimitive::U8) => {
-                    consume_type::<BitVec<Lsb0, u8>>(input)
-                }
-                TypeDef::Primitive(TypeDefPrimitive::U16) => {
-                    consume_type::<BitVec<Lsb0, u16>>(input)
-                }
-                TypeDef::Primitive(TypeDefPrimitive::U32) => {
-                    consume_type::<BitVec<Lsb0, u32>>(input)
-                }
-                TypeDef::Primitive(TypeDefPrimitive::U64) => {
-                    consume_type::<BitVec<Lsb0, u64>>(input)
-                }
-                store => {
-                    return Err(EventsDecodingError::InvalidBitSequenceType(format!(
-                        "{:?}",
-                        store
-                    ))
-                    .into())
-                }
-            }
-        }
-    }
-}
-
-/// The possible errors that we can run into attempting to decode events.
-#[derive(Debug, thiserror::Error)]
-pub enum EventsDecodingError {
-    /// Unsupported primitive type
-    #[error("Unsupported primitive type {0:?}")]
-    UnsupportedPrimitive(TypeDefPrimitive),
-    /// Invalid compact type, must be an unsigned int.
-    #[error("Invalid compact primitive {0:?}")]
-    InvalidCompactPrimitive(TypeDefPrimitive),
-    /// Invalid compact type; error details in string.
-    #[error("Invalid compact composite type {0}")]
-    InvalidCompactType(String),
-    /// Invalid bit sequence type; bit store type or bit order type used aren't supported.
-    #[error("Invalid bit sequence type; bit store type {0} is not supported")]
-    InvalidBitSequenceType(String),
-}
-
+/// Event related test utilities made use of outside this module.
 #[cfg(test)]
-mod tests {
+pub(crate) mod test_utils {
```

Comment:
I moved some common event test bits to this module so that I could use them in filter_events tests, too.